### PR TITLE
Dilpreet | Test | Integrate msg91 api to send mail on customer/vendor page

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

## Release Notes

- **New Features**
  - Added functionality to send bulk emails to vendors and customers through new buttons in the contact management interface.
  - Introduced a custom email dialog for tailored email interactions.
  - Enhanced email content suggestion retrieval based on the type of recipient (customer or vendor).
  - Added a "Save and Send" button for email submissions.

- **Bug Fixes**
  - Improved error handling and state management for email operations.

- **Documentation**
  - Updated documentation comments for clarity and consistency.

- **Localization**
  - Added new localization entries for bulk email functionality and updated existing keys in English, Hindi, and Marathi.
  - Added new entries for "Save and Send" in localization files.

### CODE REVIEW

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 1]
```diff
@@ -284,6 +284,22 @@ <h5>
                                             >
                                                 {{ localeData?.bulk_payment }}
                                             </button>
+                                            <button
+                                                mat-menu-item
+                                                *ngIf="selectedAccountsList?.length > 0 && activeTab === 'vendor'"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.html]:
```suggestion
                                                *ngIf="selectedAccountsList?.length > 0"
```

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 2]
```diff
@@ -284,6 +284,22 @@ <h5>
...
+                                                (click)="sendBulkEmail('vendor')"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.html]:
```suggestion
                                                (click)="sendBulkEmail(activeTab === 'vendor'  ? 'vendor' : 'customer')"
```

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/contact/contact.component.html]: use common function

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 3]
```diff
@@ -243,11 +245,15 @@ export class ContactComponent implements OnInit, OnDestroy {
     public voucherApiVersion: 1 | 2 = 2;
     /** True if consolidated branch */
     public isConsolidatedBranch: boolean;
+    /** Stores the send email bulk request  */
+    public sendBulkEmailRequest: SendBulkEmailTemplateRequest;
+    /** Observable for bulk email success response */
+    public bulkEmailSuccess$ = this.componentStore.select(state => state.sendBulkEmailISuccess);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]:
```suggestion
    public bulkEmailSuccess$: any = this.componentStore.select(state => state.sendBulkEmailISuccess);
```

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 4]
```diff
@@ -1655,123 +1673,39 @@ export class ContactComponent implements OnInit, OnDestroy {
...
-
-        if (lastTabType) {
-            this.translationComplete(true);
+   * Open custom email dialog
+   *
+   * @param {any} account
+   * @memberof ContactComponent
+   */
+    public openCustomEmailDialog(account: any, activeTab: string): void {
+        let data = {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]:
```suggestion
        const data = {
```

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 5]
```diff
@@ -1655,123 +1673,39 @@ export class ContactComponent implements OnInit, OnDestroy {
...
+            activeTab: activeTab,
+            accountUniqueName: account?.uniqueName
         }
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]:
```suggestion
        };
```

File: apps/web-giddh/src/app/models/api-models/Contact.ts [hunk 6]
```diff
@@ -202,4 +202,14 @@ export class BulkUpdateInvoiceCustomfields {
     customField2: string = '';
     customField3: string = '';
 }
+/**
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/models/api-models/Contact.ts:206]:
```suggestion

/**
```

File: apps/web-giddh/src/app/services/invoice.service.ts [hunk 7]
```diff
@@ -826,12 +826,21 @@ export class InvoiceService {
     /**
      * Get email content suggestions
      *
+     * @param {string} type Type of content suggestions to get ('customer' or 'vendor')
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/invoice.service.ts]:
```suggestion
     * @param {string} type of content suggestions to get ('customer' or 'vendor')
```

File: apps/web-giddh/src/app/shared/template-froala/template-froala.component.html [hunk 8]
```diff
@@ -107,10 +150,20 @@ <h3>{{ commonLocaleData?.app_custom_email_template }}</h3>
...
                 </button>
+                <button
+                    *ngIf="inputData?.activeTab"
+                    aria-label="save and send email"
+                    mat-stroked-button
+                    color="primary"
+                    type="button"
+                    (click)="onSubmit('send')"
+                >
+                    Save and Send
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/template-froala/template-froala.component.html]: Translate

File: apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts [hunk 9]
```diff
@@ -293,7 +296,7 @@ export class TemplateFroalaComponent implements OnInit {
      * @memberof TemplateFroalaComponent
      */
     public getEmailTemplates(): void {
-        this.componentStore.getAllEmailTemplate(this.voucherType);
+        this.componentStore.getAllEmailTemplate(this.inputData?.activeTab ? this.inputData?.activeTab : this.inputData);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts:300]:
```suggestion
        this.componentStore.getAllEmailTemplate(this.inputData?.activeTab ? this.inputData.activeTab : this.inputData);
```

File: apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts [hunk 10]
```diff
@@ -308,7 +311,7 @@ export class TemplateFroalaComponent implements OnInit {
      * @memberof TemplateFroalaComponent
      */
     public getEmailContents(): void {
-        this.componentStore.getEmailContentSuggestions(null);
+        this.componentStore.getEmailContentSuggestions(this.inputData?.activeTab ? this.inputData?.activeTab : this.inputData);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts]:
```suggestion
        this.componentStore.getEmailContentSuggestions(this.inputData?.activeTab ? this.inputData.activeTab : this.inputData);
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts (hunk 9 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts:300]: no need of this

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**